src/matlab2tikz.m

```diff
@@ -2930,15 +2939,13 @@ case size(cData,2) % specific y-data is given
 
             % Calculate the width of each bar and the center point shift as in
             % makebars.m
-            if isempty(m2t.barShifts)
-                % Get the shifts of the bar centers.
-                % In case of numBars==1, this returns 0,
-                % In case of numBars==2, this returns [-1/4, 1/4],
-                % In case of numBars==3, this returns [-1/3, 0, 1/3],
-                % and so forth.
-                barWidth = groupWidth/numBars; % assumption
-                m2t.barShifts = ((1:numBars) - 0.5) * barWidth - groupWidth/2;
-            end
+            % Get the shifts of the bar centers.
+            % In case of numBars==1, this returns 0,
+            % In case of numBars==2, this returns [-1/4, 1/4],
+            % In case of numBars==3, this returns [-1/3, 0, 1/3],
+            % and so forth.
+            barWidth = groupWidth/numBars; % assumption
```

Comment:
So, `h.BarWidth` is first assumed to be equal one for the determination of the `barShift`? And on line 2956 then the actual value is retrieved for the caluclation of `barWidth`?
I yes, please consider renaming the variable `barWidth` here and in the next line.

I understand that the variable was named like this previously, but while you are at it you might make the code more clear.